MiniApp/Classes/Extensions/URLSession+SessionProtocol.swift

```diff
@@ -0,0 +1,14 @@
+extension URLSession: SessionProtocol {
+    func startDataTask(with request: URLRequest, completionHandler: @escaping (Result<ResponseData, Error>) -> Void) {
+        dataTask(with: request) { (data, response, error) in
+            if let error = error {
+                return completionHandler(.failure(error))
+            }
+            guard let httpResponse = response as? HTTPURLResponse, let data = data else {
+                let error = NSError(domain: "APIClient", code: (response as? HTTPURLResponse)?.statusCode ?? 0, userInfo: [NSLocalizedDescriptionKey: "Unknown server error occurred"])
```

Comment:
It seems this could use the extension defined in `MiniAppClient.swift` or maybe add another one in there. What do you think?

Reply:
Done